fe/src/main/java/org/apache/doris/common/proc/CurrentQueryInfoProvider.java

```diff
@@ -0,0 +1,518 @@
+     * @throws AnalysisException
+     */
+    public Map<String, Consumption> getQueriesConsumptions(Collection<QueryStatisticsItem> items)
+            throws AnalysisException {
+        triggerReportAndWait(items, getWaitTime(items.size()), true);
+        final Map<String, Consumption> queryConsumpations = Maps.newHashMap();
+        for (QueryStatisticsItem item : items) {
+            queryConsumpations.put(item.getQueryId(), new Consumption(item.getQueryProfile()));
+        }
+        return queryConsumpations;
+    }
+
+    /**
+     * Return query's instances consumption.
+     *
+     * @param item
+     * @return
+     * @throws AnalysisException
+     */
+    public Collection<InstanceConsumption> getQueryInstancesConsumptions(QueryStatisticsItem item) throws AnalysisException {
+        triggerReportAndWait(item, getWaitTime(1), false);
+        final Map<String, RuntimeProfile> instanceProfiles = collectInstanceProfile(item.getQueryProfile());
+        final List<InstanceConsumption> instanceConsumptions = Lists.newArrayList();
+        for (QueryStatisticsItem.FragmentInstanceInfo instanceInfo : item.getFragmentInstanceInfos()) {
+            final RuntimeProfile instanceProfile = instanceProfiles.get(DebugUtil.printId(instanceInfo.getInstanceId()));
+            Preconditions.checkNotNull(instanceProfile);
+            final InstanceConsumption consumption =
+                    new InstanceConsumption(
+                            instanceInfo.getFragmentId(),
+                            instanceInfo.getInstanceId(),
+                            instanceInfo.getAddress(),
+                            instanceProfile);
+            instanceConsumptions.add(consumption);
+        }
+        return instanceConsumptions;
+    }
+
+    /**
+     * Profile trees is query profile -> fragment profile -> instance profile ....
+     * @param queryProfile
+     * @return instanceProfiles
+     */
+    private Map<String, RuntimeProfile> collectInstanceProfile(RuntimeProfile queryProfile) {
+        final Map<String, RuntimeProfile> instanceProfiles = Maps.newHashMap();
+        for (RuntimeProfile fragmentProfile : queryProfile.getChildMap().values()) {
+            for (Map.Entry<String, RuntimeProfile> entry: fragmentProfile.getChildMap().entrySet()) {
+                Preconditions.checkState(instanceProfiles.put(parseInstanceId(entry.getKey()), entry.getValue()) == null);
+            }
+        }
+        return instanceProfiles;
+    }
+
+    /**
+     * Instance profile key is "Instance ${instance_id} (host=$host $port)"
+     * @param str
+     * @return
+     */
+    private String parseInstanceId(String str) {
+        final String[] elements = str.split(" ");
+        if (elements.length == 4) {
+            return  elements[1];
+        } else {
+            Preconditions.checkState(false);
+            return "";
+        }
+    }
+
+    private long getWaitTime(int numOfQuery) {
+        final int oneQueryWaitTime = 200;
```

Comment:
You'd better add a suffix like 'ms' or 's' to let others know about the time unit.